endless_pagination/static/endless_pagination/js/endless-pagination.js

```diff
@@ -266,13 +265,15 @@
                     }
                 );
 
+                // compare form fields value with URL parameters
                 $('select', $( settings.formSelector )).each(
-                      function()
-                      {
-                         var selectName = $(this).attr('name');
-                         if ( !$(this).val() && data[selectName] != undefined )
-                            delete data[selectName];
-                      }
+                    function()
+                    {
+                        var selectName = $(this).attr('name');
+                        // check field value length (since we have strings and arrays)
+                        if ( $(this).val().length === 0 && data[selectName] !== undefined )
```

Comment:
If we want to remove empty lists: don't we want this to be an OR?
```suggestion
                        if ( $(this).val().length === 0 || data[selectName] !== undefined )
```

Reply:
I haven't looked at this code in forever - can you point me to an example where this  does/does not work as expected

Reply:
It's a `&&`: if the field value is empty/[] AND the URL parameter is NOT undefined, we need to clean it up.
This causes values from any input that returns a list (select2, etc...) to not be cleared form the querystring when you submit an empty form (in Liga Inan, go to Contact, Filter, set a facility or location, clear it, see the url doesn't get cleaned)  :-)

Reply:
Aha! Right